index.js

```diff
@@ -370,6 +370,10 @@ Synopsis.prototype.winners = function (n, weights) {
 
   if (!results) return
 
+  // If none of the results have a defined pinPercentage, return the empty winners list
+  const noPinPercentage = results.filter(r => typeof r.pinPercentage === 'undefined')
```

Comment:
What happens if pinPercentage is `null`

Reply:
@NejcZdovc can check for null as well, in the crash that was being investigated it seemed that this only was ever undefined.